core/server/utils/image-size-from-url.js

```diff
@@ -51,71 +48,56 @@ module.exports.getImageSizeFromUrl = function getImageSizeFromUrl(imagePath) {
             }
         }
 
-        options = url.parse(imagePath);
+        imagePath = url.parse(imagePath);
+        requestOptions = {
+            headers: {
+                'User-Agent': 'Mozilla/5.0'
+            },
+            timeout: timeout,
+            encoding: null
+        };
 
-        requestHandler = imagePath.indexOf('https') === 0 ? https : http;
-        options.headers = {'User-Agent': 'Mozilla/5.0'};
+        got(
+            imagePath,
+            requestOptions
+        ).then(function (response) {
+            try {
+                dimensions = sizeOf(response.body);
```

Comment:
could use promisify here, to make this use the promise signature  we use elsewhere, that would further reduce the number of error cases

Reply:
After trying some more time to promisify the `sizeOf` fn, I figured that it won't work asynchronously when working with the buffer. See the note [here](https://github.com/image-size/image-size#asynchronous).

So IMO we have three options:
1. Leave the whole `fn` wrapped in a promise like it is now
2. Use the URL and not the buffer (which will then again not work when reading from the local file storage as planned in #8900)
3. Make it work synchronously and just wrap it in a `try`

Reply:
I would leave this as it is. Your comment in the code base makes it very clear 👍